src/Xpath/XpathCapableInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Dhii\Data\Tree\Xpath;
+
+use Traversable;
+use Dhii\Data\Tree\NodeInterface;
+
+/**
+ */
+interface XpathCapableInterface
+{
+    /**
+     * Retrieves nodes that match a specific xPath query.
+     *
+     * @since [*next-version*]
+     *
+     * @param NodeInterface $node  The node, on which to run the xPath query.
+     * @param string        $query The query to run on the node.
```

Comment:
What do you think about accepting `string|StringableInterface`? A thought that comes to mind is a "query builder".